ext/soap/php_http.c

```diff
@@ -617,15 +617,34 @@ int make_http_soap_request(zval        *this_ptr,
 		smart_str_append_smart_str(&soap_headers, &soap_headers_z);
 
 		if (soap_version == SOAP_1_2) {
-			smart_str_append_const(&soap_headers,"Content-Type: application/soap+xml; charset=utf-8");
+			if (context &&
+		           (tmp = php_stream_context_get_option(context, "http", "content_type")) != NULL &&
+		           Z_TYPE_P(tmp) == IS_STRING) {
+				if (Z_STRLEN_P(tmp) > 0) {
```

Comment:
You'll probably want to move this check into the above if as well. Otherwise no content-type at all will be sent if "content_type" is empty. Or is that the intention?

Reply:
Right, we need a way to unset the Content-Type too.

Reply:
At least in this case this is not going to work: While the content type will be skipped, the code below will still append `; action=` etc, which will result in a broken header.

Reply:
After some thoughts on your comments, you are right, let's avoid an empty content-type. I thought it could be usefull to have such flexibilty, but the use case does not exist. Event wort, it can lead to a bad header including the action as your commented.

It is fixed by the update:
   https://github.com/php/php-src/pull/4065/commits/17578914c9dd54d13582176506c99aac844662e3

Reply:
As you already have the `Z_STRLEN_P(tmp) > 0` check in the outer if now, you can drop the inner `if (Z_STRLEN_P(tmp) > 0)` to make the code a bit simpler.